File: edukate-checker/src/main/java/io/github/sanyavertolet/edukate/checker/services/ChatService.java
package io.github.sanyavertolet.edukate.checker.services;

import io.github.sanyavertolet.edukate.checker.domain.RequestContext;
import io.github.sanyavertolet.edukate.checker.dtos.ModelResponse;
import lombok.NonNull;
import reactor.core.publisher.Mono;

public interface ChatService {
    Mono<ModelResponse> makeRequest(@NonNull RequestContext ctx);
}

File: edukate-checker/src/main/kotlin/io/github/sanyavertolet/edukate/checker/services/ChatService.kt
package io.github.sanyavertolet.edukate.checker.services

import io.github.sanyavertolet.edukate.checker.domain.RequestContext
import io.github.sanyavertolet.edukate.checker.dtos.ModelResponse
import reactor.core.publisher.Mono

fun interface ChatService {
    fun makeRequest(ctx: RequestContext): Mono<ModelResponse>
}
